contracts/pair_xyk_sale_tax/src/contract.rs

```diff
@@ -0,0 +1,1442 @@
+use std::convert::TryInto;
+use std::str::FromStr;
+use std::vec;
+
+use cosmwasm_schema::cw_serde;
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    attr, coins, from_binary, to_binary, Addr, BankMsg, Binary, CosmosMsg, Decimal, Decimal256,
+    Deps, DepsMut, Env, Fraction, MessageInfo, QuerierWrapper, Reply, ReplyOn, Response, StdError,
+    StdResult, SubMsg, SubMsgResponse, SubMsgResult, Uint128, Uint256, Uint64, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw20::{Cw20ExecuteMsg, Cw20ReceiveMsg, MinterResponse};
+
+use astroport::asset::{
+    addr_opt_validate, check_swap_parameters, format_lp_token_name, Asset, AssetInfo, CoinsExt,
+    PairInfo, MINIMUM_LIQUIDITY_AMOUNT,
+};
+use astroport::factory::PairType;
+use astroport::generator::Cw20HookMsg as GeneratorHookMsg;
+use astroport::pair::{ConfigResponse, DEFAULT_SLIPPAGE, MAX_ALLOWED_SLIPPAGE};
+use astroport::pair::{
+    CumulativePricesResponse, Cw20HookMsg, ExecuteMsg, InstantiateMsg, PoolResponse, QueryMsg,
+    ReverseSimulationResponse, SimulationResponse, TWAP_PRECISION,
+};
+use astroport::querier::{query_factory_config, query_fee_info, query_supply};
+use astroport::{token::InstantiateMsg as TokenInstantiateMsg, U256};
+use cw_utils::parse_instantiate_response_data;
+
+use crate::error::ContractError;
+use crate::state::{Config, BALANCES, CONFIG};
+use astroport::pair_xyk_sale_tax::{
+    MigrateMsg, SaleTaxConfigUpdates, SaleTaxInitParams, TaxConfigChecked,
+};
+use astroport_pair::state::{Config as XykConfig, CONFIG as XYK_CONFIG};
```

Comment:
Could you confirm the astroport-pair versions you support in migration could be deserialized with these structures?

Reply:
I added a test migrating from 1.3.1. Unfortunately 1.3.0 and 1.4.0 are not on crates.io. Can you add them?